components/plan-build/bin/hab-plan-build.sh

```diff
@@ -569,6 +569,21 @@ _ensure_origin_key_present() {
   debug "At least one signing key for $pkg_origin found in $cache"
 }
 
+# **Internal** Ensures that the plan pkg_version adheres to the semver specification.
+# Additionally, it checks to ensure that the pkg_version does not contain any dots at
+# the beginning of the string, at the end, nor multiple sequential dots anywhere within.
+# The checks for dots as just described, prevent inoperable or unexpected results when operating
+# the hab CLI or interacting with Builder HTTP APIs as the dots have special meaning in a URI.
+_validate_pkg_version() {
+  local inoperable_re="^\.+|\.$|\.{2,}"
+  local wellformed_re="^([[[:digit:]]\.]*[[:digit:]]+)(.+)?"
+
+  if [[ ! $pkg_version =~ $wellformed_re ]] || [[ $pkg_version =~ $inoperable_re ]]; then
+     exit_with "Found pkg_version with an invalid format, aborting! See: https://semver.org/"
```

Comment:
If we don't accept _all_ semver strings, then we should be _very_ clear about that in this user-facing error message. Otherwise, that will only lead to confusion and frustration.